### ISSUE TITLE
MicroG Login Loop: "Account Already Exists"

### ISSUE THREAD

**Issue by jmessy919** [microg/GmsCore#1403]
Running latest Lineage OS for MicroG on my rooted OnePlus 6T. I cannot get past the add account screen even after properly authenticating. It acts as if I never added an account to MicroG. When I try again, I get the "this account already exists on your device."

I have searched through previous issues, even so far back as 2018 with no answers. Any assistance is appreciated. Thanks.

**Comment by jmessy919** [comment]
On top of this, now when I switch back and forth between apps and go back to YouTube Vanced, it pops up the "Add Account" screen. So definitely worse off than before I tried adding an account to MicroG.

**Comment by AinzTheSupremeOne** [comment]
Looks like the same problem as the one mentioned in https://github.com/microg/GmsCore/issues/1373 . Please we need a fix. We can't even switch to another account from Google apps.

**Comment by jmessy919** [comment]
@AinzTheSupremeOne I would say it might be related, but this is across the board, in the MicroG core settings also. It's not specifically a Vanced issue.

**Comment by Oizaro** [comment]
> @AinzTheSupremeOne I would say it might be related, but this is across the board, in the MicroG core settings also. It's not specifically a Vanced issue.

The bug report #1373 describes a behavioir of stock apps, not of Vanced.

**Comment by jmessy919** [comment]
> The bug report #1373 describes a behavioir of stock apps, not of Vanced.

Sure, but if you look at the issue it does not describe what I expressed above. My YouTube loads, but I cannot login.

**Comment by Oizaro** [comment]
> > The bug report #1373 describes a behavioir of stock apps, not of Vanced.
> 
> Sure, but if you look at the issue you and @AinzTheSupremeOne referenced, it does not describe the bug which I expressed above. My YouTube loads, but I cannot login. MicroG acts like there is no account, even when I definitely entered one. It then tells me that my account already exists without actually logging in.

In truth this bug prevents the login on every newer app (starting from January update).

So yeah, is a general bug.

**Comment by jmessy919** [comment]
> In truth this bug prevents the login on every newer app (starting from January update).
> 
> So yeah, is a general bug (what I would mean).

Fair enough, thanks for the clarification. I do hope this gets resolved soon!

**Comment by MijoHuer** [comment]
Ir happens the same with me. Even microg can't reconize that there is an account in my device, but the google ads works perfectly

**Comment by erik9012** [comment]
Is there something planned to try to fix the bug? because its since January. Is known how the bug can appear?

**Comment by ShubhamPatilsd** [comment]
Hey everyone so I think I might have fixed this issue. I simply uninstalled the GMSCore and GMSProxy apps and reinstalled them and now it works like a charm!

**Comment by ale5000-git** [comment]
Everything should already be fixed on the latest version of microG GmsCore so I close it for now.
Open a new issue if needed.

**Comment by erik9012** [comment]
Which version is the fix?

**Comment by ale5000-git** [comment]
microG GmsCore 0.2.23.214816

**Comment by sayz** [comment]
Same problem here. I have the latest versions microg and YouTube vanced.

**Comment by ale5000-git** [comment]
The problem of "Account Already Exists" happens because the account really exist, specially on non clean systems (so you won't see it inside Android accounts).

The solution is delete these files (if they exist):
/data/system/users/0/accounts.db
/data/system_de/0/accounts_de.db
/data/system_ce/0/accounts_ce.db

IMPORTANT: This will remove all accounts on the device (even non Google ones).

After this you can add the account normally.

**Comment by Connorgilbert1** [comment]
I am experiencing this same issue, with the latest version of micro g.
Couldn't find those files in my device

**Comment by ale5000-git** [comment]
@Connorgilbert1

Here is the full list of files to delete (somw devices have some files, while some other have other files):
```
/data/system_de/0/accounts_de.db
/data/system_de/0/accounts_de.db-journal

/data/system_ce/0/accounts_ce.db
/data/system_ce/0/accounts_ce.db-journal

/data/system/accounts.db
/data/system/accounts.db-journal

/data/system/users/0/accounts.db
/data/system/users/0/accounts.db-journal

/data/system/sync/stats.bin
/data/system/sync/pending.xml
/data/system/sync/accounts.xml
/data/system/sync/status.bin
```

**Comment by riekusr** [comment]
I am on android 13 and can't seem to find any of those files. I have micro G on the work profile and logged in with work gmail. Everytime I am geeting the loop "account already exists" when it asks to login.

**Comment by ale5000-git** [comment]
The list is for various versions of Android so it is normal that you haven't all of them but you must have some.
You need root to see them and if you are in recovery you must also mount /data to see them.

**Comment by rokejulianlockhart** [comment]
This continues for me. Latest version of CalyxOS, installed today.

**Comment by ale5000-git** [comment]
@rokejulianlockhart
Please open a new issue, and specify if the account doesn't appear at all in Android settings or if it appears and are just some apps that doesn't see it.

**Comment by tom282f3** [comment]
Today I ran into the same issue after I logged out every devices for my Google account. Always getting "Account Already Exists" in MicroG. I tried cleaning data for the MicroG, reinstalling it, even using my cell phone number to log in. All failed.
Luckily for me I realized MicroG account was registered in Android Settings. I removed the account in the Android Settings and everything is fine now.
Not a custom Android though. I am running Android 1p on Galaxy Note 9.
